sdk/src/main/java/io/opentelemetry/sdk/trace/export/BatchSpanProcessor.java

```diff
@@ -277,28 +258,26 @@ private void exportBatches(ArrayList<ReadableSpan> spanList) {
     }
 
     // Exports the list of SpanData to the SpanExporter.
+    @SuppressWarnings("BooleanParameter")
     private void onBatchExport(final List<SpanData> spans) {
-      Future<?> submission =
-          executorService.submit(
+      if (exportAvailable.compareAndSet(true, false)) {
+        try {
+          final CompletableResultCode result = spanExporter.export(spans);
+          result.thenRun(
               new Runnable() {
                 @Override
                 public void run() {
-                  // In case of any exception thrown by the service handlers catch and log.
-                  try {
-                    spanExporter.export(spans);
-                  } catch (Throwable t) {
-                    logger.log(Level.WARNING, "Exception thrown by the export.", t);
+                  if (!result.isSuccess()) {
+                    logger.log(Level.FINE, "Exporter failed");
```

Comment:
this makes me think that we probably should think about adding an Exception/Throwable to the CompletableResultCode, but that isn't necessary for this PR.

Reply:
The contract is for an exporter to either succeed or fail as per the result code. In the case of failure, it could be worth indicating why, with an optional throwable as a parameter. I'd say that it is up to the export class itself to determine what should constitute being worthy of logging here though, as failure may well be "normal" and I don't think we want to pollute the logs.